File: Neon Number/NEON.java
/*A neon number is a number where the sum of digits of square of the number is equal to the number. 
For example if the input number is 9, its square is 9*9 = 81 and sum of the digits is 9. i.e. 9 is a neon number.*/

import java.io.*;
import java.util.*;

class NEON {
	static int sumOfDigitsOfSquare(int num)throws Exception {
		int sum = 0;
		num = num * num;
		while(num != 0) {
			sum += num % 10;
			num /= 10;
		}
		return  sum;
	}
	public static void main(String[] args) {
		try {
			BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
			int[] input = Arrays.stream(br.readLine().split(" ")).mapToInt(Integer::valueOf).toArray();
			for (int i = 0; i < input.length; i++) {
				if (sumOfDigitsOfSquare(input[i]) == input[i]) {
					System.out.print(input[i] + " ");
				}
			}
		}
		catch(Exception e) {
			return ;
		}
	}
}

File: Neon Number/NEON.kt
/**
 * @created October 2, 2019
 * A neon number is a number where the sum of digits of square of the number is equal to the number.
 * For example if the input number is 9, its square is 9*9 = 81 and sum of the digits is 9. i.e. 9 is a neon number.
 */

class NEON {

    fun solve() {
        println("Input a number to check if it is a NEON number: ")
        val number = readLine()!!.toInt()
        if (neonNumber(number) == number) {
            println("$number is a NEON number.")
        } else {
            println("$number is not a NEON number.")
        }
    }

    private fun neonNumber(number: Int): Int {
        return sumDigits(number * number)
    }

    private fun sumDigits(number: Int): Int {
        var sum = 0
        var temporal = number
        while (temporal != 0) {
            sum += temporal % 10
            temporal /= 10
        }
        return sum
    }

}

fun main() {
    NEON().solve()
}